core/dbt/parser/manifest.py

```diff
@@ -1141,6 +1141,24 @@ def process_metrics(self, config: RuntimeConfig):
 
     def process_saved_queries(self, config: RuntimeConfig):
         """Processes SavedQuery nodes to populate their `depends_on`."""
+        # Note: This will also capture various nodes which have been re-parsed
+        # because they refer to some other changed node, so there will be
+        # false positives. Ideally we would compare actual changes.
+        semantic_manifest_changed = False
+        SemanticManifestNode = Union[SavedQuery, SemanticModel, Metric]
+        semantic_manifest_nodes: chain[SemanticManifestNode] = chain(
+            self.manifest.saved_queries.values(),
+            self.manifest.semantic_models.values(),
+            self.manifest.metrics.values(),
+        )
+        for node in semantic_manifest_nodes:
```

Comment:
Nit: This can be simplified to

```
if any(node.created_at > self.started_at for node in semantic_manifest_nodes):
    return
```

Reply:
The logic of "any" is backward, we want to execute the process metrics for node code instead of returning, but doing it that way would make merging into dbt-mantle more muddy because it executes a different function, so I'm going to leave this as it is.